### DISCUSSION TITLE
Error Writing Screenshot

### DISCUSSION THREAD

**Discussion by smp-marco** [smplayer-dev/smplayer#569] [Q&A] [answered]
Version: 22.7.0 (revision 10091) (64-bit)
Using Qt 5.15.2 (compiled with Qt 5.15.2)
Using MPV 0.34.0

Tried a screenshot, but it wasn't available.
Checked preferences, and saw that the target directory no longer existed.
I browsed to an existing directory, and applied.

Attempted a screenshot, and got:
"Error Writing Screenshot"

Closed SMP and re-launched.
Same problem.

Screenshots always worked previously.
My gut feeling is that something went wrong when I attempted a screenshot to a dir that didn't exist.

I haven't yet tried a reinstall because I have my GUI all setup :)

Not the end of the world to reinstall, but I figured it would be best to report the issue first.
:)

**Comment by smplayer-dev** [comment]
Are you using a snap or flatpak package?

**Reply by smp-marco** [reply]
Probably not, because I don't know what those terms mean.
BTW the pc has been rebooted, and I'm still getting the same error message.

**Comment by smplayer-dev** [comment]
Check the permissions of the folder where you want to save the screenshots.

**Reply by smp-marco** [reply]
It's a default folder (I have tried writing to a different folder).
It stores screenshot images from a week ago.

I found that a third iteration of a video was needed.
Consequently, I needed to firm up the naming strategy.
Hence, I renamed two of the folder groups, specifically renaming the sub directories with unique names.
This is why the screenshot path was pointing to a directory that didn't exist.

Just had a thought, and checked ... can screenshot capture from a youtube url?

**Reply by smp-marco** [reply]
I have found a web hosted video, and loaded it's url into SMP.
The screenshot works!

Consequently, I presume the problem lies with the youtube connection.

Downloading the videos from YT, and loading them from file, enables screenshot.
However, it would be more efficient, if the screenshot could be taken from a YT url video.

Can this be fixed, or is it not allowed by YT?

**Comment by smplayer-dev** [comment, marked as answer]
This is because mpv tries to use part of the video url as the filename, but it's too long and fails. You can for example replace the screenshot template (preferences -> general) to `cap_%n_%p_%02n` and it will use a number instead of a filename and it will work.

**Comment by smp-marco** [comment]
That works great Ricardo.
Thanks for taking the time to help :)
